.husky/pre-commit

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+pnpm run pre-commit
+git add --all
```

Comment:
if someone is trying to just commit one file or one small change, won't this auto-commit all changes? I think yeah probably just want to do `pnpm run check-format`. I'd much rather error out of a commit than have all my changes auto committed without my intent.

Reply:
Ah, thank you for this catch, I looked into this a bit more and found [lint-staged](https://www.npmjs.com/package/lint-staged), which only applies the the linting/type-checking/formatting to **staged** files, runs the jobs in parallel, and only displays output on error (I'm sure it can be configured if we want warnings as well). Paired with Husky this is the optimal DX afaict. It's clean, configurable, and catches mistakes up front very nicely.

Here's an example of an error output with the following code:
<img width="329" alt="Screenshot 2023-10-20 at 12 19 27 PM" src="https://github.com/MutinyWallet/mutiny-web/assets/38222767/97e322d1-e0d8-426d-bcc0-e30b8cd23c4d">

<img width="1506" alt="Screenshot 2023-10-20 at 12 21 45 PM" src="https://github.com/MutinyWallet/mutiny-web/assets/38222767/f0eb7734-1ccd-43b5-ab07-469e69ea77f9">

A non-error output:
<img width="335" alt="Screenshot 2023-10-20 at 1 00 11 PM" src="https://github.com/MutinyWallet/mutiny-web/assets/38222767/5aee4740-91ec-4929-a15d-4b625e31c4de">

Reply:
yeah I think this is a good improvement. now it just needs to work! I'm getting a bunch of wild output, do you know why this would be? (this commit should be good, I only added a comment)

<img width="1039" alt="Screenshot 2023-10-20 at 4 30 40 PM" src="https://github.com/MutinyWallet/mutiny-web/assets/543668/e9a522f1-30c7-40e8-b8c8-6a026443db32">

Reply:
I wasn't getting this output for some reason until I reran `pnpm install`. Weird. Anyway the solution ended up having to do with typescript not being able to find the `tsconfig.json` when invoked from husky, but when we run it as bash it can: `bash -c 'pnpm run check-types'`. Should be fixed now.

More here: https://github.com/lint-staged/lint-staged/issues/825#issuecomment-674575655